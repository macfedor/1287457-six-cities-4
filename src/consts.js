export const PlaceType = {
  APARTMENT: `apartment`,
  ROOM: `room`,
  HOUSE: `house`,
  HOTEL: `hotel`
};

export const CardType = {
  CITY: `city`,
  NEAR: `near`,
};

export const CardClassName = {
  [CardType.CITY]: `cities__place-card`,
  [CardType.NEAR]: `near-places__card`,
};

export const ImageWrapperClassName = {
  [CardType.CITY]: `cities__image-wrapper`,
  [CardType.NEAR]: `near-places__image-wrapper`,
};

export const maxNearbyOffers = 3;

export const maxReviews = 10;

export const monthNames = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `Octoder`,
  `November`,
  `December`,
];
