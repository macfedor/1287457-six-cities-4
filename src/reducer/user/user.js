import {extend} from "../../utils/common";
import history from "../../history";
import {AppRoute} from "../../consts";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  userEmail: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_USER_EMAIL: `SET_USER_EMAIL`,
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        if (response.status === 200) {
          dispatch(ActionCreator.requiredAuthorization(AuthorizationStatus.AUTH));
          dispatch(ActionCreator.setUserEmail(response.data.email));
        }
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password,
    })
    .then((response) => {
      dispatch(ActionCreator.requiredAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.setUserEmail(response.data.email));
    })
    .then(() => {
      return history.push(AppRoute.ROOT);
    })
    .catch((err) => {
      throw err;
    });
  },
};

const ActionCreator = {
  requiredAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status
  }),
  setUserEmail: (result) => ({
    type: ActionType.SET_USER_EMAIL,
    payload: result
  }),

};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
    case ActionType.SET_USER_EMAIL:
      return extend(state, {
        userEmail: action.payload
      });
    default:
      return state;
  }
};

export {reducer, Operation, ActionType, ActionCreator, AuthorizationStatus};
