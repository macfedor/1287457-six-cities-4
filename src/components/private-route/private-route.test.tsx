import * as React from "react";
import * as renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route";
import {MemoryRouter} from "react-router";
import {AppRoute} from "../../consts";
import {AuthorizationStatus} from "../../reducer/user/user";

const MockComponent = () => <div />;

it(`should render PivateRoute with mock component`, () => {
  const tree = renderer.create(
      <MemoryRouter initialEntries={[AppRoute.FAVORITES]}>
        <PrivateRoute
          exact
          path={AppRoute.FAVORITES}
          authorizationStatus={AuthorizationStatus.AUTH}
          render={() => <MockComponent />}
        />
      </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
