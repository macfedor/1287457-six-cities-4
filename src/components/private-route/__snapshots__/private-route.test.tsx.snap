// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PivateRoute with mock component 1`] = `<div />`;
